plugins/modules/monitoring/monit.py

```diff
@@ -182,15 +182,15 @@ def _parse_status(self, output, err):
             return status
 
     def is_process_present(self):
-        rc, out, err = self.module.run_command('%s summary %s' % (self.monit_bin_path, self.command_args), check_rc=True)
+        rc, out, err = self.module.run_command([self.monit_bin_path, 'summary', self.command_args], check_rc=True)
         return bool(re.findall(r'\b%s\b' % self.process_name, out))
 
     def is_process_running(self):
         return self.get_status().is_ok
 
     def run_command(self, command):
         """Runs a monit command, and returns the new status."""
-        return self.module.run_command('%s %s %s' % (self.monit_bin_path, command, self.process_name), check_rc=True)
+        return self.module.run_command([self.monit_bin_path, command, self.process_name], check_rc=True)
```

Comment:
```suggestion
        return self.module.run_command([self.monit_bin_path] + command + [self.process_name], check_rc=True)
```
and make sure that `command` is always a list (right now it's in half the cases).

Reply:
Yes, it is quite confusing. But the plot thickens: there are many occurrences of calls to `self.module.run_command()`, but `self.run_command()` is called only twice, on lines 237 and 250. Indeed one was passing a list and the other a string. For the sake of simplicity, I think we should keep the string version of the method, rather than moving to a list one. The moethod name could change though, but will skip that for now. @snopoke maybe that's an improvement you could work on?